garnett/gsdhoomdfilewriter.py

```diff
@@ -123,46 +123,21 @@ def write(self, trajectory, stream):
                 except AttributeError:
                     types = ['A']
                 snap.particles.types = types
-                try:
-                    snap.particles.typeid = [types.index(typeid) for typeid in frame.types]
-                except AttributeError:
-                    pass
-                try:
-                    snap.particles.position = frame.positions
-                except AttributeError:
-                    pass
-                try:
-                    snap.particles.orientation = frame.orientations
-                except AttributeError:
-                    pass
-                try:
-                    snap.particles.velocity = frame.velocities
-                except AttributeError:
-                    pass
-                try:
-                    snap.particles.mass = frame.mass
-                except AttributeError:
-                    pass
-                try:
-                    snap.particles.charge = frame.charge
-                except AttributeError:
-                    pass
-                try:
-                    snap.particles.diameter = frame.diameter
-                except AttributeError:
-                    pass
-                try:
-                    snap.particles.moment_inertia = frame.moment_inertia
-                except AttributeError:
-                    pass
-                try:
-                    snap.particles.angmom = frame.angmom
-                except AttributeError:
-                    pass
-                try:
-                    snap.particles.image = frame.image
-                except AttributeError:
-                    pass
+                for prop in [
+                        'position',
+                        'orientation',
+                        'velocity',
+                        'mass',
+                        'charge',
+                        'diameter',
+                        'moment_intertia',
+                        'angmom',
+                        'image'
+                        ]:
```

Comment:
Since a trajectory object is being passed to this function, this line can be further simplified as
```python
for  prop in trajectory.TRAJ_ATTRIBUTES[4:]:
```

Reply:
Resolved in 7ac0f68